File: spring-security/src/main/java/com/example/springsecex/util/StringUtil.java
package com.example.springsecex.util;

import java.util.Locale;
import java.util.Objects;

public final class StringUtil {

    private StringUtil() {
    }

    public static boolean isEmpty(String text) {
        return Objects.isNull(text) || text.isEmpty();
    }

    public static boolean nonEmpty(String text) {
        return Objects.nonNull(text) && !text.isEmpty();
    }

    public static long toLong(String value, long def) {
        if (isEmpty(value)) {
            return def;
        }

        try {
            return Long.valueOf(value);
        } catch (NumberFormatException e) {
            return def;
        }
    }

    public static int toInt(String value, int def) {
        if (isEmpty(value)) {
            return def;
        }

        try {
            return Integer.valueOf(value);
        } catch (NumberFormatException e) {
            return def;
        }
    }

    public static String toLowerCase(String src) {
        return null == src ? null : src.toLowerCase(Locale.getDefault());
    }

    public static String toUpperCase(String src) {
        return null == src ? null : src.toUpperCase(Locale.getDefault());
    }

    public static String toString(Object obj) {
        return obj == null ? null : obj.toString();
    }
}

File: spring-web-kotlin/src/main/kotlin/com/example/springwebex/util/StringUtil.kt
package com.example.springwebex.util

import java.util.Locale

object StringUtil {

    fun isEmpty(text: String?): Boolean {
        return text == null || text.isEmpty()
    }

    fun nonEmpty(text: String?): Boolean {
        return text != null && text.isNotEmpty()
    }

    fun toLong(value: String?, def: Long): Long {
        if (isEmpty(value)) return def
        return try {
            value!!.toLong()
        } catch (e: NumberFormatException) {
            def
        }
    }

    fun toInt(value: String?, def: Int): Int {
        if (isEmpty(value)) return def
        return try {
            value!!.toInt()
        } catch (e: NumberFormatException) {
            def
        }
    }

    fun toLowerCase(src: String?): String? {
        return src?.lowercase(Locale.getDefault())
    }

    fun toUpperCase(src: String?): String? {
        return src?.uppercase(Locale.getDefault())
    }

    fun toString(obj: Any?): String? {
        return obj?.toString()
    }
}
